packages/react/src/bags/hooks/types/useBag.ts

```diff
@@ -1,18 +1,18 @@
+import type { Bag } from '@farfetch/blackout-client/src/bags/types';
 import type { BagItemHydrated } from '@farfetch/blackout-redux/src/entities/types';
-import type { Error } from '@farfetch/blackout-client/types';
-import type { NormalizedBag } from '@farfetch/blackout-redux/src/bags/types';
+import type { State } from '@farfetch/blackout-redux/src/bags/types';
 
 export type UseBag = () => {
-  bag: NormalizedBag | null;
-  bagId: NormalizedBag['id'] | null;
-  error: Error | null | undefined;
-  fetchBag: () => Promise<NormalizedBag>;
-  hasItems: boolean | undefined;
-  isLoading: boolean | undefined;
+  bag: State['result'];
+  id: State['id'];
+  error: State['error'] | undefined;
+  fetchBag: () => Promise<Bag>;
+  isEmpty: boolean | undefined;
+  isLoading: State['isLoading'] | undefined;
```

Comment:
I think this can never be `undefined`, not only because the `isBagLoading` selector is always a boolean, but also because your `isLoading` constant is actually a "this OR that", so it's always a boolean (and since this is not just a proxy for the `isBagLoading` selector, I'd use `boolean` instead of `State['isLoading']` 🤔)

Reply:
You are right. I removed `undefined`.